src/console/dlt-control.c

```diff
@@ -112,6 +113,27 @@ typedef struct {
 } DltReceiveData;
 
 
+void wait_send_buffer_empty(int fd)
+{
+#ifdef __linux__
+    struct timespec ts = {0};
+    /* Wait until send buffer is empty */
+    while (1)
+    {
+        int unsent;
+        ioctl(fd, TIOCOUTQ, &unsent);
+        if (!unsent)
+        {
+            break;
+        }
+
+        ts.tv_sec = 0;
+        ts.tv_nsec = 100000 * 1000;
```

Comment:
Maybe it's worth to use NANOSEC_PER_MILLISEC or NANOSEC_PER_SEC which are defined in src/console/dlt-control-common.h. Also, it's better to have these code outside of while() if it doesn't change dynamically.

Reply:
done